File: apps/collector/src/models/impact.py
"""Impact assessment Pydantic model — DB entity (impact_assessments table)."""
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field


class ImpactAssessment(BaseModel):
    """DB entity: impact_assessments table (update × company_profile impact score)."""
    id: UUID
    regulatory_update_id: UUID
    company_profile_id: UUID
    impact_score: int = Field(..., ge=0, le=100)
    impact_category: str  # direct | indirect | monitoring
    reasoning: str
    recommended_actions: list[str] | None = None
    created_at: datetime | None = None

    model_config = {"from_attributes": True}


class ImpactAssessmentCreate(BaseModel):
    """Payload for creating an impact assessment."""
    regulatory_update_id: UUID
    company_profile_id: UUID
    impact_score: int = Field(..., ge=0, le=100)
    impact_category: str
    reasoning: str
    recommended_actions: list[str] | None = None

File: packages/shared/src/types/impact.ts
/**
 * Impact assessment — company profile vs regulatory update
 */
export type ImpactCategory = 'direct' | 'indirect' | 'monitoring';

export interface ImpactAssessment {
  id: string;
  regulatory_update_id: string;
  company_profile_id: string;
  impact_score: number; // 0-100
  impact_category: ImpactCategory;
  reasoning: string;
  recommended_actions: string[];
  created_at: string;
}

export interface CompanyProfile {
  id: string;
  user_id: string;
  company_name?: string;
  license_types: string[];
  jurisdictions: string[];
  domains: string[];
  entity_size: string;
  services: string[];
  created_at: string;
  updated_at: string;
}
